### ISSUE TITLE
Beetle PSX Crashes on launch after masOS system update

### ISSUE THREAD

**Issue by SquareMan** [libretro/beetle-psx-libretro#884]
[Crash Report.txt](https://github.com/libretro/beetle-psx-libretro/files/11402957/Crash.Report.txt)
## Description

It seems that the most recent MacOS software update has broken something with Beetle PSX. Retroarch now crashes immediately on starting any content with it. Crash Report attached.

### Expected behavior

Selected content to launch

### Actual behavior

RetroArch crashes

### Steps to reproduce the bug

1. Open a PS1 game with Beetle PSX (Tested with Final Fantasy VII and Final Fantasy VIII)

### Bisect results

N/A: This same version of RetroArch/Beetle PSX worked before performing the system update.

### Version/Commit

[If you're using RetroArch, this information is on the bottom left after loading the core. You can do this via Main Menu -> Load Core]

- Beetle PSX: 0.9.44.1 fd812d4

Please specify whether you are using Beetle PSX or Beetle PSX HW.

Beetle PSX

### Core options

[List all the core options you have changed]
- Skip BIOS

Please also specify, if reporting a graphical glitch for Beetle PSX HW, your "Renderer" core option setting and your frontend selected video driver.

### Environment information

- Frontend: Retroarch 1.15.0
- OS: macOS Ventura 13.3.1
- Compiler: N/A
- GPU: N/A

**Comment by warmenhoven** [comment]
Is this still an issue?

**Comment by warmenhoven** [comment]
Yes, I can reproduce this. I've tried all four graphics drivers and they all do the same thing, in basically the same place inside Metal but taking different routes to get there.

**Comment by warmenhoven** [comment]
Interestingly both the HW and non-HW cores crash similarly.

**Comment by warmenhoven** [comment]
If I run in Xcode with "Metal Shader Validation" turned on, it doesn't crash and doesn't give any warnings.

**Comment by warmenhoven** [comment]
And it works fine on latest iOS, so that's not going to give any clues either.

**Comment by warmenhoven** [comment]
Hm, there are other issues with this core as well: the recompiler seems to cause it to hang indefinitely at start; and it can't be dlopen()'ed a second time.

**Comment by dixius99** [comment]
I get a very similar crash report. Seems to crash no matter what video driver I use. Interestingly though: for me, this problem only began some time after around February 15. I had used the Beetle PSX core previously, and it worked fine.

[Crash Report--2024-05-03.txt](https://github.com/libretro/beetle-psx-libretro/files/15205277/Crash.Report--2024-05-03.txt)

**Comment by LibretroAdmin** [comment]
MoltenVK issue
